File: src/main/java/hzt/aoc/day16/Field.java
package hzt.aoc.day16;

import hzt.aoc.Pair;

import java.util.ArrayList;
import java.util.List;

public class Field {

    private static int next = 0;
    private final int nr;
    private final String fieldName;
    private final List<Pair<Integer, Integer>> valueRanges = new ArrayList<>();

    public Field(final String fieldName) {
        this.nr = next++;
        this.fieldName = fieldName;
    }

    public void addRange(final Pair<Integer, Integer> range) {
        if (range != null) {
            valueRanges.add(range);
        }
    }

    public boolean containsValueInRanges(final Integer integer) {
        for (final Pair<Integer, Integer> p : valueRanges) {
            if (integer >= p.getLeft() && integer <= p.getRight()) {
                return true;
            }
        }
        return false;
    }

    public int getNr() {
        return nr;
    }

    public String getFieldName() {
        return fieldName;
    }

    public static void setNext(final int next) {
        Field.next = next;
    }

    @Override
    public String toString() {
        return "Field{" +
                "nr=" + nr +
                ", fieldName='" + fieldName + '\'' +
                ", valueRanges=" + valueRanges +
                '}';
    }
}

File: src/main/kotlin/hzt/aoc/day16/Field.kt
package hzt.aoc.day16

import hzt.aoc.Pair

class Field(fieldName: String) {

    private val nr: Int = next++
    private val fieldName: String
    private val valueRanges: MutableList<Pair<Int, Int>> = ArrayList()

    fun addRange(range: Pair<Int, Int>) {
        valueRanges.add(range)
    }

    fun containsValueInRanges(integer: Int): Boolean {
        for (p in valueRanges) {
            if (integer >= p.left && integer <= p.right) {
                return true
            }
        }
        return false
    }

    override fun toString(): String {
        return "Field{" +
                "nr=" + nr +
                ", fieldName='" + fieldName + '\'' +
                ", valueRanges=" + valueRanges +
                '}'
    }

    companion object {
        private var next = 0
        fun setNext(next: Int) {
            Companion.next = next
        }
    }

    init {
        this.fieldName = fieldName
    }
}
